src/components/profile/connections/ConnectionBox.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div class="connection-box" :style="{ 'background-color': color }">
+<div class="header">
+  <img draggable="false" :class="{ 'account-profile': true, round: data.connected }" :src="data.connected ? data.account.avatar : getIcon(icon)" />
+  <div class="account-info">
+    <span class="account-name">{{ data.connected ? data.account.user : displayName }}</span>
+    <br>
+    <span v-if="data.connected" class="connection-name">{{ displayName }}</span>
+  </div>
+  <!-- TODO: Fix button to the right -->
+  <button v-if="!data.connected" class="button is-pulled-right" @click="openConnectWindow(data)">CONNECT</button>
+</div>
+    <div v-if="data.connected" class="content">
+      <slot class="config"></slot>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'ConnectionBox',
+  props: ['color', 'displayName', 'connected', 'icon', 'data'],
+  methods: {
+    openConnectWindow (data) {
+      this.$parent.$parent.openConnectWindow(data)
+    },
+    getIcon (icon) {
+      return require(`../../../assets/logos/${icon}`)
+    }
+  }
+}
+</script>
+
+<style>
+  .connection-box {
+    border-radius: 5px;
+  }
+  .connection-box > .header {
+    background: rgba(0, 0, 0, .1);
+    border-radius: 5px 5px 0px 0px;
+    padding: 16px;
+    border-bottom: .6px solid rgba(63, 63, 63, .27);
+    align-items: center;
+    display: -webkit-box;
+    display: -ms-flexbox;
+    display: flex;
+  }
+  .connection-box > .header > .account-profile {
+    width: 64px;
+    height: 64px;
+  }
+  .connection-box > .header > .round {
+    border-radius: 128px;
+  }
+  .connection-box > .header > .account-info {
```

Comment:
Don't use too much nesting, learn about `BEM`

Reply:
i wanted to use scoped but that doesnt work

Reply:
done